CMakeLists.txt

```diff
@@ -785,20 +802,24 @@ endif ()
 
 
 ## src/ executables
-foreach (prog main test test-would-substitute test-size-params test-buffer-serialize hb-ot-tag)
-  set (prog_name ${prog})
-  if (${prog_name} STREQUAL "test")
-    # test can not be used as a valid executable name on cmake, lets special case it
-    set (prog_name test-test)
-  endif ()
-  add_executable(${prog_name} ${PROJECT_SOURCE_DIR}/src/${prog}.cc)
-  target_link_libraries(${prog_name} harfbuzz ${THIRD_PARTY_LIBS})
-endforeach ()
-set_target_properties(hb-ot-tag PROPERTIES COMPILE_FLAGS "-DMAIN")
+# as harfbuzz should be built with add_executable for Emscripten builds, it is not
```

Comment:
It should be possible to build Harfbuzz as a static library with Emscripten, though I wonder if the use case here was something that wanted to do different? In general I'd recommend having the same static library vs shared library vs executable structures on all platforms that CMake is targeting, and, let's say if Emscripten wants to do an executable, then do it in a way that other platforms can also build executables, i.e. decouple the target type from the platform. Or was there some specific issue here?

Reply:
This was just experiment to see what we else do here, removed.